src/libraries/System.Net.Security/src/System/Net/Security/SslStream.cs

```diff
@@ -888,5 +944,35 @@ private static void ThrowNotAuthenticated()
         {
             throw new InvalidOperationException(SR.net_auth_noauth);
         }
+
+        internal sealed unsafe class PoolingPointerMemoryManager : MemoryManager<byte>
```

Comment:
Can you add a comment about why this is necessary (and the other options we considered and dismissed for now)?